src/deployment.ts

```diff
@@ -87,6 +87,32 @@ export class Deployment {
         return Promise.resolve();
     }
 
+    @vsCommand('openshift.deployment.scale')
+    static async scalingPod(component: KubernetesObject): Promise<void> {
+        if (!component) {
+            return;
+        }
+        let count = '1';
+        count = await inputValue(`How many replicas would you like to scale ${component.kind}/${component.metadata.name}?`,
+            count,
+            false,
+            (value: string) => {
+                const trimmedValue = value.trim();
+                if (trimmedValue.length === 0) {
+                    return 'Scale value cannot be empty';
+                }
+                if (parseInt(trimmedValue, 10) < 0) {
+                    return 'Scale value should greater than zero';
+                }
+            },
+            'Scale Count'
+        );
+        const response = await Oc.Instance.scalePod(component.metadata.name, count);
+        if (response.indexOf('scaled') !== -1) {
+            OpenShiftExplorer.getInstance().refresh(component);
```

Comment:
If you are increasing the number of replicas (eg. 1 --> 3), then the new pods don't show up when you refresh the tree here, since the pods take a while to reach the "Running" state. I think this can be a bit confusing for the user.

However, I can't think of a way to address this other than refreshing the Application Explorer on a timer, which we are trying to avoid, so I think this is good for now.

Reply:
Refreshing the tree view based on `'Pod is Running'` text on terminal view. Now we can get the updated pods